contracts/StablecoinConverter.sol

```diff
@@ -237,12 +238,13 @@ contract StablecoinConverter is EpochTokenLocker {
         uint16[] memory tokenIdsForPrice
     ) public returns (uint256) {
         require(acceptingSolutions(batchIndex), "Solutions are no longer accepted for this batch");
+        burnPreviousAuctionFees();
```

Comment:
I'm just curious, is there a reason for this being placed here? It feels kind of unnatural to me, but then again I am a noob at smart contract coding and want to learn :smiley:.

Reply:
My train of thought when putting it here was that it we should at least verify that the solution being submitted it relevant for the current batch. Now that I think about it, we would probably also want  to check the claimed objective value first. Technically, whatever changes are made during this transaction will revert if any of the requirements are not satisfied. I am now considering moving this to the end where all of the state information is actually updated. Nice! I hope this helps with the learning.

Reply:
I would vote for pushing it down by one line
Reasoning: first make all the solution submission gas saving tests: acceptingSolutions && claimObjectiveValue> currentObjective value, and then we start with

Reply:
> and then we start with...

The suspense is killing me!

Reply:
I have moved just below the first group of `requirements` (for readability).